### DISCUSSION TITLE
Can I use treesitter to intercept parse errors to correct the error?

### DISCUSSION THREAD

**Discussion by kendalldita** [tree-sitter/tree-sitter#3285] [Q&A] [answered]
I have not yet tried setting up treesitter. I have documents which conform to a grammar, I want to parse the documents, assume that parse errors are only because of missing elements (this will be true). Then supply default versions of what is missing.

Is that something there is support for, with treesitter?

**Comment by WillLillis** [comment, marked as answer]
This is certainly something you could build on top of tree-sitter! You could parse a document with tree-sitter, use the query api on the resulting tree to find error nodes (and their corresponding position within the document), and then (via your own implementation, as tree-sitter doesn't support that specific functionality) insert the corresponding default values based on the node's parents, siblings, or some other context.
